app/models/vm_or_template.rb

```diff
@@ -51,7 +51,7 @@ class VmOrTemplate < ApplicationRecord
   validates_presence_of     :name, :location
   validates_inclusion_of    :vendor, :in => VENDOR_TYPES.values
 
-  has_one                   :miq_server, :foreign_key => :vm_id
+  has_one                   :miq_server, :foreign_key => :vm_id, :inverse_of => :vm
```

Comment:
I think `template.miq_server` is invalid...so as @matthewd suggests, I think the `has_one                   :miq_server, :foreign_key => :vm_id, :inverse_of => :vm` belongs in vm.rb, not vm_or_template.rb.  Can you research where this was originally added?  I don't see why template would have it.